zipkin-lens/src/components/DependenciesPage/DependenciesPage.test.js

```diff
@@ -0,0 +1,146 @@
+        <button
+          key={nodeName}
+          type="button"
+          data-testid={`${nodeName}-button`}
+          onClick={() => objectHighlighted({ type: 'node', getName: () => nodeName })}
+        >
+          {nodeName}
+        </button>
+      ))
+    }
+  </div>
+)));
+
+describe('<DependenciesPage />', () => {
+  const exampleDependencies = [
+    { parent: 'serviceA', child: 'serviceB', callCount: 1000 },
+    { parent: 'serviceB', child: 'serviceC', callCount: 3200 },
+    { parent: 'serviceB', child: 'serviceD', callCount: 200 },
+    { parent: 'serviceD', child: 'serviceE', callCount: 500 },
+    { parent: 'serviceE', child: 'serviceF', callCount: 1000 },
+    { parent: 'serviceC', child: 'serviceE', callCount: 2000 },
+    { parent: 'serviceC', child: 'serviceG', callCount: 1200 },
+  ];
+
+  const commonProps = {
+    isLoading: false,
+    dependencies: [],
+    fetchDependencies: () => {},
+    clearDependencies: () => {},
+    location: {
+      search: '?endTs=1571979713227&startTs=1571893313227',
+    },
+    history: { push: () => {} },
+  };
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  it('should render a loading indicator when isLoading is true', () => {
+    const { queryAllByTestId } = render(<DependenciesPageImpl {...commonProps} isLoading />);
+    const loadingIndicators = queryAllByTestId('loading-indicator');
+    expect(loadingIndicators.length).toBe(1);
+  });
+
+  it('should render a explain box if there is not graph yet', () => {
+    const { queryAllByTestId } = render(
+      <DependenciesPageImpl {...commonProps} dependencies={[]} />,
+    );
+    const explainBoxes = queryAllByTestId('explain-box');
+    expect(explainBoxes.length).toBe(1);
+  });
+
+  it('should render a dependencies graph when isLoading is false and there is a graph data', () => {
+    const { queryAllByTestId } = render(
+      <DependenciesPageImpl
+        {...commonProps}
+        dependencies={exampleDependencies}
+      />,
+    );
+    const dependenciesGraphs = queryAllByTestId('dependencies-graph');
+    expect(dependenciesGraphs.length).toBe(1);
+  });
+
+  it('should render a node detail when nodes are clicked', () => {
+    const { getByTestId, queryAllByTestId } = render(
+      <DependenciesPageImpl
+        {...commonProps}
+        dependencies={exampleDependencies}
+      />,
+    );
+    const serviceA = getByTestId('serviceA-button');
+    fireEvent.click(serviceA);
+    let nodeDetails = queryAllByTestId('node-detail');
+    expect(nodeDetails.length).toBe(1);
+
+    // And remove the node detail when graph's background is clicked.
+    const graphBackground = getByTestId('graph-background');
+    fireEvent.click(graphBackground);
+    nodeDetails = queryAllByTestId('node-detail');
+    expect(nodeDetails.length).toBe(0);
+  });
+
+  it('should fetch dependencies when mounted by using query parameters', () => {
```

Comment:
by is redundant with using in this case

```suggestion
  it('should fetch dependencies when mounted using query parameters', () => {
```